pkg/gcv/cf/cf.go

```diff
@@ -81,41 +78,49 @@ func (cf *ConstraintFramework) validateTemplate(t *configs.ConstraintTemplate) e
 	return err
 }
 
-// AddTemplate tracks an additional constraint template. This template is only used if a constraint is provided.
-func (cf *ConstraintFramework) AddTemplate(template *configs.ConstraintTemplate) error {
-	if _, exists := cf.templates[template.GeneratedKind]; exists {
-		return status.Errorf(codes.AlreadyExists, "Conflicting constraint templates with kind %s from file %s", template.GeneratedKind, template.Confg.FilePath)
-	}
-	if err := cf.validateTemplate(template); err != nil {
-		return status.Error(codes.InvalidArgument, err.Error())
+// Configure will set the constraint templates and constraints for ConstraintFramework
+func (cf *ConstraintFramework) Configure(templates []*configs.ConstraintTemplate, constraints []*configs.Constraint) error {
```

Comment:
I'm unsure if changing to a single method for configuring CF is desirable. Three concerns:
1. It tightly couples adding constraints and constraint templates, which might be separate (particularly in test cases)
2. I'd assume this means we're drifting further from Gatekeeper? In k8s-based systems, constraints could be added incrementally instead of scanning an entire filesystem at once.
3. We're losing the persistence of templates/constraints within FCV which will make it harder to interrogate violations for additional metadata.

Reply:
1) See NewValidator, we load all of them at once anyway, this is way way easier than building up internal structures inside of the OPA storage.Store
2) This will get thrown out in favor of the real Constraint Framework which handles all that stuff.
3) If we already do this, it should still work, if we need to add it, we can put it back in.

Keep in mind this implementation (everything in this file) is going to get thrown in the trash when we integrate with constraint framework.  Once the switch over comes, changing things to fit the new interface won't be much of an issue, but implementing things to work properly here is going to be a lot of work for not much benefit since it's all going to get deleted.

Reply:
Ok, this might end up breaking Scorecard but if so we can fix separately.